fragmenter/fragment.py

```diff
@@ -19,6 +19,113 @@
 
 
+def expand_states(molecule, protonation=True, tautomers=False, stereoisomers=True, consider_aromaticity=True, maxstates=200,
+                  verbose=True, filename=None):
+    """
+    Expand molecule states (choice of protonation, tautomers and/or stereoisomers).
+    Protonation states expands molecules to protonation of protonation sites (Some states might only be reasonable in
+    very high or low pH. ToDo: Only keep reasonable protonation states)
+    Tatutomers: Should expand to tautomer states but most of hte results are some resonance structures. Defualt if False
+    for this reason
+    Stereoisomers expands enantiomers and geometric isomers (cis/trans).
+    Returns set of SMILES
+
+    Parameters
+    ----------
+    molecule: OEMol
+        Molecule to expand
+    protonation: Bool
+        If True will enumerate protonation states. Default True
+    tautomers: Bool
+        If True, will enumerate tautomers. Default False (Note: This is False because results usually give resonance structures
+        which ins't needed for torsion scans
+    stereoisomers: Bool
+        If True will enumerate stereoisomers (cis/trans and R/S). Default is True
+    consider_atomaticity: Bool
+    maxstates: int
+        Defulat 200
+    verbose: Bool
+    filename: str
+        Filename to save SMILES to. If None, SMILES will not be saved to file. Defualt None.
+
+    Returns
+    -------
+    states: set of SMILES for enumerated states
+
+    """
+    title = molecule.GetTitle()
+    states = set()
+    molecules = [molecule]
+    if verbose:
+        logger().info("Enumerating states for {}".format(title))
+    if protonation:
+        molecules.extend(_expand_states(molecules, enumerate='protonation', consider_aromaticity=consider_aromaticity,
+                                   maxstates=maxstates, verbose=verbose))
+    if tautomers:
+        molecules.extend(_expand_states(molecules, enumerate='tautomers', consider_aromaticity=consider_aromaticity,
+                                   maxstates=maxstates, verbose=verbose))
+    if stereoisomers:
+        molecules.extend(_expand_states(molecules, enumerate='stereoisomers', maxstates=maxstates, verbose=verbose))
+
+    if filename:
+        oname = filename
+        ofs = oechem.oemolostream()
+        if not ofs.open(oname):
+            oechem.OEThrow.Fatal('Unable to open {} for writing molecules.'.format(oname))
+
+    for molecule in molecules:
+        states.add(fragmenter.utils.create_mapped_smiles(molecule, tagged=False, explicit_h=False))
+    if filename:
+        count = 0
+        smiles_list = []
+        for molecule in states:
+            molecule = molecule + ' ' + title + '_' + str(count)
+            count += 1
+            smiles_list.append(molecule)
+        fragmenter.utils.to_smi(smiles_list, filename)
+
+    return states
+
+
+def _expand_states(molecules, enumerate='protonation', consider_aromaticity=True, maxstates=200, verbose=True):
```

Comment:
It's important that we still document private methods since there are multiple developers working on this project. Can you be sure to add docstrings for private methods too?

Reply:
Done.